stdlib/http/src/main/ballerina/src/http/cookie/cookieStore.bal

```diff
@@ -0,0 +1,343 @@
+            foreach var cookie in self.allSessionCookies {
+                if (!((url.startsWith("https") && cookie.secure) || cookie.secure == false)) {
+                    return cookiesToReturn;
+                }
+                if (!((url.startsWith("http") && cookie.httpOnly) || cookie.httpOnly == false)) {
+                    return cookiesToReturn;
+                }
+                if (cookie.hostOnly == true) {
+                    if (cookie.domain == domain && checkPath(path, cookie)) {
+                        cookiesToReturn[cookiesToReturn.length()] = cookie;
+                    }
+                } else {
+                    if ((domain.endsWith("." + cookie.domain) || cookie.domain == domain ) && checkPath(path, cookie)) {
+                        cookiesToReturn[cookiesToReturn.length()] = cookie;
+                    }
+                }
+            }
+            // TODO:Get persistent cookies from database.
+            return cookiesToReturn;
+        }
+    }
+
+     # Gets all the session and persistent cookies.
+     #
+     # + return - All the session and persistent cookies
+     public function getAllCookies() returns Cookie[] {
+         lock {
+             // TODO:Get persistent cookies.
+             return self.allSessionCookies;
+         }
+     }
+
+     # Removes a specific cookie.
+     #
+     # + name - Name of the cookie to be removed
+     # + domain - Domain of the cookie to be removed
+     # + path - Path of the cookie to be removed
+     # + return - Return true if the relevant cookie is removed, false otherwise
+     public function removeCookie(string name, string domain, string path) returns boolean {
+         lock {
+             // Removes the session cookie in the cookie store which is matched with  given name, domain and path.
+             int k = 0;
+             while (k < self.allSessionCookies.length()) {
+                 if (name == self.allSessionCookies[k].name && domain == self.allSessionCookies[k].domain && path ==  self.allSessionCookies[k].path) {
+                     int j = k;
+                     while (j < self.allSessionCookies.length()-1) {
+                         self.allSessionCookies[j] = self.allSessionCookies[j + 1];
+                         j = j + 1;
+                     }
+                     Cookie lastCookie = self.allSessionCookies.pop();
+                     return true;
+                 }
+                 k = k + 1;
+             }
+             // TODO:Remove from the database if it is a persistent cookie.
+             return false;
+         }
+     }
+
+    # Removes all expired cookies.
+    public function removeExpiredCookies() {
+        lock {
+            // TODO:If expired, remove those persistent cookies from the database.
+        }
+    }
+
+    # Removes all the session and persistent cookies.
+    public function clear() {
+        lock {
+            // TODO:Remove all persistent cookies from the database.
```

Comment:
Do we need todo here? Shall we create issues for not completed stuff? Check all the places

Reply:
Created issues:  #19796